minigunapp/tasks.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from celery import shared_task
+from minigunapp.models import Email
+from django.core.mail import send_mail
+
+
+@shared_task
+def send_email(id):
+    """ Task that attempts to send the email with the given id.
+    Retries on a delay until max failures reached.
+    """
+
+    email = Email.objects.get(id=id)
+
+    try:
+        send_mail(
+            email.subject,
+            email.body,
+            email.from_,
+            [email.to],
+            fail_silently=False,
+        )
+        email.status = 'sent'
+        logging.info("Sent email {0}".format(email.id))
+    except Exception as ex:
+        logging.error("Failed to send email {0} ({1}): {2}"
+                      .format(email.id, email.retries. str(ex)))
+        email.status = 'error'
+        email.error_message = str(ex)
+        if(email.retries < 3):
+            logging.info("Queueing email {0} for retry".format(email.id))
+            email.retries += 1
+            email.save()
```

Comment:
Actually in this case of the db going away and being unable to save, the exception causes the task not to be retried since apply_async won't be called again.